### PR TITLE
.travis.yml: update to use xs-opam

### PR DESCRIPTION
**Pull request by mseri**
This will also test the build against opam+xs-opam.

### CODE REVIEW

**Comment by gaborigloi** [comment]
Looks good, however, do we need testing with the `EXTRA_REMOTES`, wouldn't that be very error-prone as we have no control over which package versions get installed?

**Comment by mseri** [comment]
Coverage tests are randomly failing, but seems not our issue.
The EXTRA_REMOTES can fail and will not invalidate the build. Travis will only consider the BASE_REPO build for the tick. It is good for us to see what is failing, if anything, if we change upstream libraries

**Comment by mseri** [comment]
Builds passed! https://travis-ci.org/mseri/message-switch/builds/213886926

File: .travis.yml [hunk 1]
```diff
@@ -2,10 +2,19 @@ language: c
...
     - OCAML_VERSION=4.02
     - PACKAGE=message-switch
-    - PINS="shared-block-ring:git://github.com/mirage/shared-block-ring"
-    - COV_CONF="./configure"
+  matrix:
+    - BASE_REMOTE=git://github.com/xapi-project/xs-opam XS_COV=1 COV_CONF="./configure"
+    - EXTRA_REMOTES=git://github.com/xapi-project/xs-opam
+matrix:
+    fast_finish: true
+    allow_failure:
```
Comment (gaborigloi) [inline .travis.yml]: I think this is `alow_failures` (plural): https://docs.travis-ci.com/user/customizing-the-build#Rows-that-are-Allowed-to-Fail

```
~/s/message-switch> travis lint .travis.yml
Warnings for .travis.yml:
[x] in matrix section: unexpected key allow_failure, dropping
```

**Pull request merged**